### ISSUE TITLE
Did Text Notification Display Work With Android 4.4.2?

### ISSUE THREAD

**Issue by miband2lover** [Freeyourgadget/Gadgetbridge#805]
#### Before opening an issue please confirm the following:
- [ yes] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [ yes] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [ yes] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:
Text is displayed only during an incoming call. All options for send text notifications on the debug menu displays only icons (and vibrates).

Already downgraded to fw 1.0.1.34 and no success either.

Must inform that in the first install of the mi fit app, fw 1.0.1.50 was installed. Then mi fit was uninstalled and gadgetbridge was installed. In this first try text on incoming calls didn't work too.

Maybe text notifications doesnt work with this old Android version?

#### Your wearable device is:
Mi band 2, hw 0.1.3.2, fw 1.0.1.39, with Mili_pro.ft and ft.en installed in this order. Android and gadgetbridge language: english.

#### Your android version is:
4.4.2 Motorola razr i
#### Your Gadgetbridge version is:
0.20.0 gadgetbridge

[gadgetbridgelog.zip](https://github.com/Freeyourgadget/Gadgetbridge/files/1287632/gadgetbridgelog.zip)

**Comment by ashimokawa** [comment]
Did you enable textual notifications in mi band settings in Gadgetbridge?

**Comment by miband2lover** [comment]
Yep! "Text Notifications - Needs firmware >= 1.0.1.28 and Mili_pro.ft* installed" ON. 

With this option off there is no text display on incaming calls, only phone icon.

**Comment by miband2lover** [comment]
Yep! "Text Notifications - Needs firmware >= 1.0.1.28 and Mili_pro.ft* installed" ON.

With this option off there is no text display on incoming calls, only phone icon.

**Comment by miband2lover** [comment]
More information:

md5sum of Mili_pro_fw, Mili_pro.ft and Mili_pro.ft.en are the same for the 1.0.1.39 version informed on this page: https://github.com/Freeyourgadget/Gadgetbridge/wiki/Mi-Band-Firmware-Update.

Must inform that while the gadgetbridges's installation notification of Mili_pro.ft and Mili_pro.ft.en finishes with a message of success (complete), installation of Mili_pro.fw doesn't. The android status bar installation progress notification continues (and never ends unless installation of Mili_pro.ft starts). 

When Mili_pro.fw installation is almost finishing (progress notification on android status bar almost in the end) there is a gadgetbridge notification that the band needs to wait to reconnect and, after, a notification of band connection, but doesn't have a notification of successful instalation of Mili_pro.fw. But the fw version informed on gadgetbridge after reconnection is the version installed.

**Comment by miband2lover** [comment]
Just some pictures to help understand what is happening during the following options of the debug menu: 

incoming call,
![img_20170910_104853_283](https://user-images.githubusercontent.com/31765000/30250176-3f548d58-9620-11e7-97e0-74ba3a4f9b4a.jpg)
generic_sms,
![img_20170910_104924_763](https://user-images.githubusercontent.com/31765000/30250177-3f573738-9620-11e7-9d76-88e6d43be085.jpg)
facebook.
![img_20170910_104943_642](https://user-images.githubusercontent.com/31765000/30250178-3f5c59f2-9620-11e7-8cee-9dbacaa2ab9d.jpg)

**Comment by danielegobbetti** [comment]
I wonder if this is the same issue as reported in 754, and your miband 2 is not receiving the second message.

**Comment by ashimokawa** [comment]
> When Mili_pro.fw installation is almost finishing (progress notification on android status bar almost in the end) there is a gadgetbridge notification that the band needs to wait to reconnect and, after, a notification of band connection, but doesn't have a notification of successful instalation of Mili_pro.fw. But the fw version informed on gadgetbridge after reconnection is the version installed.

Interesting. Same behaviour as on the Amazfit Bip, I thoght that was new...

**Comment by miband2lover** [comment]
On the facebook, signal and others debug options, the band displays first this icon of the last picture, and after the icon of the second picture.

But just the icons, no text.

**Comment by miband2lover** [comment]
![img_20170910_134931_918](https://user-images.githubusercontent.com/31765000/30251562-37b4ffb6-9638-11e7-8075-ba34c3d74d49.jpg)

Solved, and I will tell how:

1- I did read the issue 754, thanks danielegobbetti.
2- Reflashed fw 1.0.1.34. Same behavior.
3- Reflashed fw 1.0.1.39. Same behavior.
4- Reflashed fonts. Same behavior.
5- Disconnected the band in gadgetbridge.
5- Reinstalled mi fit, signed in, mi fit upgraded fw to version 1.0.1.53.
6- Unpaired band in mifit.
7- Connected the band in gadgetbridge.
8- Tested all debug option in gadgetbridge, now only unknown and generic_calendar options doesn't shows the message icon with "Test" text under it. Finally t worked.
9- Uninstalled mi fit.
10- Reflashed fw 1.0.1.39. Dind't reflashed the font files. It still works.

Hungouts notifications (and gadgetbridge's "Create test notification") still doesn't show the second icon with the text under it ,but I think this is the normal behavior, right? Are there any odds of hungouts notifications show text too?

It seems that the mi fit app did something to the band. What do ya think?

**Comment by danielegobbetti** [comment]
Perhaps each firmware has specific font file to flash and any misalignment is causing these issues.
I wonder if we can detect the currently installed font version and give a warning.

**Comment by miband2lover** [comment]
So did you think mi fit upgraded fw 1.0.1.53 install corrected, somehow, the misalignment? Because in worked in this version, and even after I flashed fw 1.0.1.39, without the font files, it is still working. 

Fw 1.0.1.39 is the last gadgetbridge sleep data compatible, so I prefer it.

**Comment by cpfeiffer** [comment]
The nextt Gadgetbridge version will support sleep data even with newer firmware version, so you could have another try.

**Comment by mpiatka** [comment]
Text notifications did not work for me on fw 1.0.1.69. Downgrading to 1.0.1.53 fixed it for me.

Fonts were installed by the Mi Fit app

- MiBand 2, hw 0.1.3.3
- Android 8.0 (HTC 10)
- Gadgetbridge 0.24.5

**Comment by ashimokawa** [comment]
I will test this weekend and enable text display upon connect. Might be that new firmwares have a different default.
